### PR TITLE
Harmony: StateGroup

### LINKED ISSUE #954: Harmony: StateGroup

### PR DESCRIPTION
**Pull request by DenisVorop**
## PR includes

- [x] Feature

## Related issues

Resolve #954 

Related issues: 
- https://github.com/taskany-inc/issues/pull/2419

## QA Instructions, Screenshots, Recordings
  
Снимок экрана 2024-03-11 в 17 44 09
Снимок экрана 2024-03-11 в 17 44 13

### CODE REVIEW

**Review by awinogradov — CHANGES_REQUESTED** [review]
We have same code for UserGroup component. It would be better to create meta component and use it in UserGroup, StateGroup and etc.

**Comment by DenisVorop** [comment]
Снимок экрана 2024-03-12 в 16 43 11

File: src/harmony/Group/Group.module.css [hunk 1]
```diff
@@ -0,0 +1,15 @@
+.Group {
+    display: flex;
+}
+
+.Group_hovered .GroupItem:hover {
```
Comment (awinogradov) [inline src/harmony/Group/Group.module.css:5]: `.Group_interactive .GroupItem:hover`

File: src/harmony/Group/Group.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+interface GroupProps {
+    children?: ReactNode;
+    className?: string;
+
+    /**
+     * Prop to determine whether to enable the adjusted stacking order of group items during hover within the group.
+     * If passed, the hovered item will be brought to the front, overlapping others.
+     */
+    withHover?: boolean;
```
Comment (awinogradov) [inline src/harmony/Group/Group.tsx:19]: `interactive` to be consistent with same behaviour in other components

**Review by awinogradov — CHANGES_REQUESTED** [review]

**Review by awinogradov — APPROVED** [review]

**Pull request merged**